islandora-indexing-triplestore/src/main/java/ca/islandora/alpaca/indexing/triplestore/processors/ProcessorUtils.java

```diff
@@ -0,0 +1,155 @@
+
+import static ca.islandora.alpaca.indexing.triplestore.processors.FcrepoHeaders.FCREPO_BASE_URL;
+import static ca.islandora.alpaca.indexing.triplestore.processors.FcrepoHeaders.FCREPO_IDENTIFIER;
+import static ca.islandora.alpaca.indexing.triplestore.processors.FcrepoHeaders.FCREPO_URI;
+import static java.util.Arrays.stream;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+import static org.apache.camel.support.ExchangeHelper.getMandatoryHeader;
+import static org.apache.jena.util.URIref.encode;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.List;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.NoSuchHeaderException;
+
+import org.slf4j.Logger;
+
+/**
+ * Utility functions for fcrepo processor classes
+ * @author Aaron Coburn
+ * @since November 14, 2014
+ *
+ * @author whikloj
+ *   Copied from fcrepo-camel and modified for use in Alpaca - 2021-10-13
+ */
+
+public final class ProcessorUtils {
+
+    private static final Logger LOGGER  = getLogger(ProcessorUtils.class);
+
+    /**
+     * This is a utility class; the constructor is off-limits.
+     */
+    private ProcessorUtils() {
+    }
+
+    private static String trimTrailingSlash(final String path) {
+        if (path.endsWith("/")) {
+            return path.substring(0, path.length() - 1);
+        }
+        return path;
+    }
+
+    /**
+     * Extract the subject URI from the incoming exchange.
+     * @param exchange the incoming Exchange
+     * @return the subject URI
+     * @throws NoSuchHeaderException when the CamelFcrepoBaseUrl header is not present
+     */
+    public static String getSubjectUri(final Exchange exchange) throws NoSuchHeaderException {
+        final String uri = exchange.getIn().getHeader(FCREPO_URI, "", String.class);
+        if (uri.isEmpty()) {
+            LOGGER.trace("uri isEmpty");
+            final String base = getMandatoryHeader(exchange, FCREPO_BASE_URL, String.class);
+            final String path = exchange.getIn().getHeader(FCREPO_IDENTIFIER, "", String.class);
+            LOGGER.trace("base is {}, path is {}", base, path);
+            return trimTrailingSlash(base) + path;
+        }
+        LOGGER.trace("Returning {}", uri);
+        return uri;
+    }
+
+    /**
+     * Create a DELETE WHERE { ... } statement from the provided subject
+     *
+     * @param subject the subject of the triples to delete.
+     * @param namedGraph an optional named graph
+     * @return the delete statement
+     */
+    public static String deleteWhere(final String subject, final String namedGraph) {
+        final StringBuilder stmt = new StringBuilder("DELETE WHERE { ");
+
+        if (!namedGraph.isEmpty()) {
+            stmt.append("GRAPH <");
+            stmt.append(encode(namedGraph));
+            stmt.append("> { ");
+        }
+
+        stmt.append("<");
+        stmt.append(encode(subject));
```

Comment:
PMD's complaining of a few things here:

```
1 	/home/adam/repos/Alpaca/islandora-indexing-triplestore/src/main/java/ca/islandora/alpaca/indexing/triplestore/processors/ProcessorUtils.java 	99 	Avoid appending characters as strings in StringBuffer.append.
2 	/home/adam/repos/Alpaca/islandora-indexing-triplestore/src/main/java/ca/islandora/alpaca/indexing/triplestore/processors/ProcessorUtils.java 	99 	StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.
3 	/home/adam/repos/Alpaca/islandora-indexing-triplestore/src/main/java/ca/islandora/alpaca/indexing/triplestore/processors/ProcessorUtils.java 	100 	StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.
```

Linking to:
* https://pmd.github.io/pmd-6.21.0/pmd_rules_java_performance.html#appendcharacterwithchar
* https://pmd.github.io/pmd-6.21.0/pmd_rules_java_performance.html#consecutiveappendsshouldreuse

Reply:
Resolved in https://github.com/Islandora/Alpaca/pull/82/commits/5e5d11e3292a7c6458442aacf47b018476640f8a